lib/mutant/isolation/fork.rb

```diff
@@ -0,0 +1,67 @@
+module Mutant
+  class Isolation
+    # Isolation via the fork(2) systemcall.
+    #
+    # We do inject so many globals and common patterns to make this unit
+    # specifiable without mocking the globals and more important: Not having
+    # mutations that bypass mocks into a real world side effect.
+    class Fork < self
+      include Anima.new(:process, :stderr, :stdout, :io, :devnull, :marshal)
+
+      # Prevent mutation from process.fork in #call to call Kernel#fork
+      undef_method :fork
+
+      # Call block in isolation
+      #
+      # @return [Object]
+      #   returns block execution result
+      #
+      # @raise [Error]
+      #   if block terminates abnormal
+      def call(&block)
+        io.pipe(binmode: true) do |reader, writer|
+          parent(reader, writer, &block)
+        end
+      rescue => exception
+        raise Error, exception
+      end
+
+      # Handle parent process
+      #
+      # @param [IO] reader
+      # @param [IO] writer
+      #
+      # @return [undefined]
+      def parent(reader, writer, &block)
+        pid = process.fork do
+          child(reader, writer, &block)
+        end
+
+        writer.close
+        marshal.load(reader)
+      ensure
+        process.waitpid(pid) if pid
+      end
+
+      # Handle child process
+      #
+      # @param [IO] reader
+      # @param [IO] writer
+      #
+      # @return [undefined]
+      def child(reader, writer)
+        result = devnull.call do |null|
+          stderr.reopen(null)
+          stdout.reopen(null)
+          yield
+        end
+
+        reader.close
+        writer.binmode
+        writer.syswrite(marshal.dump(result))
```

Comment:
You *could* do `writer.binmode.syswrite(...)` if you wanted. I noticed that rubocop flagged this method due to too many expressions. I don't know what the limit is, but maybe this will get you under the threshold.